### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by luxuriant777**

### CODE REVIEW

File: app/players/dwarves/dwarf.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from abc import ABC
+
+from app.players.player import Player
+
+
+class Dwarf(Player, ABC):
```
Comment (Yaroslav-Stetsko) [inline app/players/dwarves/dwarf.py:6]: Dwarf should be inherited from Player 
Also no need to import ABC in dwarf module

File: app/players/elves/elf.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from abc import ABC
+
+from app.players.player import Player
+
+
+class Elf(Player, ABC):
```
Comment (Yaroslav-Stetsko) [inline app/players/elves/elf.py:6]: Elf should be inherited from Player 
Also no need to import ABC in elf module

File: app/players/elves/druid.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            favourite_spell: str) -> None:
+        super().__init__(musical_instrument, nickname)
+        self._favourite_spell = favourite_spell
+
+    def get_rating(self) -> int:
+        return len(self._favourite_spell)
+
+    def player_info(self) -> str:
+        return f"Druid {self.nickname}. {self.nickname} has " \
+               f"a favourite spell: {self._favourite_spell}"
```
Comment (Yaroslav-Stetsko) [inline app/players/elves/druid.py:19]: use () when returning multiple string 
delete \
use everywhere

**Review by Yaroslav-Stetsko — CHANGES_REQUESTED** [review]

File: app/players/dwarves/dwarf_blacksmith.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from app.players.dwarves.dwarf import Dwarf
+
+
+class DwarfBlacksmith(Dwarf):
+
+    def __init__(
+            self,
+            favourite_dish: str,
+            nickname: str,
+            skill_level: int) -> None:
```
[the remark this answers is not in the document]
Comment (TPolina) [reply app/players/dwarves/dwarf_blacksmith.py:10]: Fix it everywhere

**Review by TPolina — CHANGES_REQUESTED** [review]

File: app/players/dwarves/dwarf.py (hunk 1 above)
Comment (TPolina) [reply app/players/dwarves/dwarf.py:6]: Both variants are ok. We can inherit ABC for explicity

File: app/players/elves/druid.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from app.players.elves.elf import Elf
```
[the remark this answers is not in the document]
Comment (TPolina) [reply app/players/elves/druid.py:1]: absolute path is better

[the remark this answers is not in the document]
Comment (Yaroslav-Stetsko) [reply app/players/elves/druid.py:1]: Got it!

File: app/main.py [hunk 6]
```diff
@@ -1 +1,16 @@
-# write your code here
+def calculate_team_total_rating(
+        player_list: list) -> int:
+    total_rating = 0
+    for player_ in player_list:
```
[the remark this answers is not in the document]
Comment (luxuriant777) [reply app/main.py]: changed

File: app/players/dwarves/dwarf_blacksmith.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (luxuriant777) [reply app/players/dwarves/dwarf_blacksmith.py]: fixed

**Review by TPolina — APPROVED** [review]

**Pull request closed without merging**